ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -343,6 +334,11 @@ static inline int get_dynamic_win_info(uint64_t remote_addr, ompi_osc_ucx_module
     int insert = -1;
     int ret;
 
+    bool lock_needed = ompi_osc_need_acc_lock(module, ompi_comm_rank(module->comm));
+    if (lock_needed) {
+        ompi_osc_ucx_lock(MPI_LOCK_SHARED, ompi_comm_rank(module->comm), 0, win);
```

Comment:
Is this the right lock? It seems like this should be the accumulate lock, like in `start_atomicity`?

Reply:
Since inside the win attach, we are locking OSC_UCX_STATE_LOCK_OFFSET offset, I think we need to try to get the same lock from the process that is calling get_dynamic_info. Thoughts?

Reply:
I was just wondering: does `ompi_osc_ucx_lock` take the same lock as `MPI_Win_lock`? The name sounds like it... In that case we'd still run into a deadlock if the calling process already owns an MPI-level locks on the window.

Reply:
yes, when I fixed the typo of the value of the target lock on get_dynamic_info, now I get an error complaining that I am trying to get a lock that was already got acquired (MPI_ERR_RMA_SYNC). It seems like MPI_Win_lock ends up locking the same lock as ompi_osc_ucx_lock. In this case, we need to use ACC lock for both window attach and get_dynamic_window

Reply:
@devreal I made the above changes. Please take a look.